docs/tutorials/shop-page.md

```diff
@@ -0,0 +1,396 @@
+	end
+end
+```
+
+Then, we simply use `Roact.Event.Activated` to tell Roact to connect the function to the _Activated_ event.
+
+```lua
+return Roact.createElement("ImageButton", {
+	BackgroundTransparency = 1,
+	Image = "",
+	LayoutOrder = order,
+	[Roact.Event.Activated] = self.onActivated,
+}, {
+	...
+}
+```
+> **Why is there brackets `[]` here?**
+>
+> In lua, the keys in a table can be any value. The most common way to map keys and values in a table construction is with `foo = something`. In this case, `foo` will be considered a string. So if we had this entry in a table `t`, we can access its value with `t.foo` or `t["foo"]`.
+>
+> What if we want a key like `"foo.bar"`? For this, the lua syntax allow us to write this entry as `["foo.bar"] = something`. So between the brackets, any value can be given. That is why brackets are use for `Roact.Event.Activated`: we don't want to pass a string that could collide with other props, instead we use a unique value owned by Roact.
+
+## Adding Animations
+
+To make our UI more lively, we are going to use the [*TweenService*](https://developer.roblox.com/en-us/api-reference/class/TweenService) to create a new Tween object. The goal is to make the icon bigger when the mouse is inside the button. To be able to do that, we need a reference to the roblox instance we want to animate. Roact can provide that through refs. First we start by creating the ref in the [`init`](https://roblox.github.io/roact/guide/state-and-lifecycle/#lifecycle-methods) lifecycle method.
+
+```lua
+function ProductItem:init()
+	self.ref = Roact.createRef()
+	...
+end
+```
+
+Since we want to animate the size of the icon, the ref is going to be in the ImageLabel props.
+
+```lua
+Icon = Roact.createElement("ImageLabel", {
+	AnchorPoint = Vector2.new(0.5, 0.5),
+	BackgroundTransparency = 1,
+	Image = image,
+	Position = UDim2.new(0.5, 0, 0.5, 0),
+	Size = UDim2.new(1, -PADDING, 1, -PADDING),
+	[Roact.Ref] = self.ref,
+}),
+```
+
+Next step is to create the Tween object when the component is mounted. The [`didMount`](https://roblox.github.io/roact/guide/state-and-lifecycle/#lifecycle-methods) lifecycle method is perfectly suited for this case, since it will be called once the instance object is created.
+
+```lua
+function ProductItem:didMount()
+	local tweenInfo = TweenInfo.new(0.2)
+	local icon = self.ref:getValue()
+
+	self.toBigIcon = TweenService:Create(icon, tweenInfo, {
+		Size = UDim2.new(1, 0, 1, 0),
+	})
+	self.toNormalIcon = TweenService:Create(icon, tweenInfo, {
+		Size = UDim2.new(1, -PADDING, 1, -PADDING),
+	})
+end
+```
+
+Don't forget to clean up! When the component will be unmounted, we need to destroy the Tween objects that were created. We use the [`willUnmount`](https://roblox.github.io/roact/guide/state-and-lifecycle/#lifecycle-methods) lifecycle method to destroy the objects.
+
+```lua
+function ProductItem:willUnmount()
+	self.toBigIcon:Destroy()
+	self.toNormalIcon:Destroy()
```

Comment:
Do Tweens actually have to be destroyed? Given that they're only stored in memory they should be cleaned up regardless when ProductItem is unmounted. They also don't seem like they have a Destroy method

Reply:
I don't think you can find the info on the [devhub](https://developer.roblox.com/en-us/api-reference/class/Tween). It does not show the ancestor functions and I don't really understand how I am suppose to know what are the ancestors of the class I am looking at. In fact, it's not really clear if it's an Instance

Anyway, you can check the ancestors on [rodocs](https://rodocs.github.io/mini/#Tween) and yes apparently it's a descendant of Instance.

But should we still destroy it? I wanted to have an example of clean up you can do in the `willUnmount` lifecycle method.